### PR TITLE
Change confirmation message if user changes email

### LINKED ISSUE #402: Improve notification on email address change
E-mail address change requires user to verify access to the new address. For this confirmation email is sent and user profile is updated only after new address confirmation.

The message that user receives in the UI on email address update is just

Updated successfully

The message is misleading. The message should say

en: Confirmation link was sent to new email address. Please confirm the address for the change to take an effect!

et: Palun kinnita eposti aadressi muutus klõpsates uuele aadressile saadetud ekirjas olevale kinnituse lingile.

### PR DESCRIPTION
**Pull request by yulgolem**
Closes #402 , adds localised notification if email was changed.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -104,4 +104,9 @@ def set_user
   def authorize_user
     authorize! :manage, @user
   end
+
+  def update_notice
+    email_changed = @user.email != update_params[:email]
```
Comment (artur-beljajev) [inline app/controllers/users_controller.rb]: Did you consider `user.email_changed?` from Rails?

File: test/system/users/edit_user_test.rb [hunk 2]
```diff
@@ -27,7 +27,7 @@ def test_update_of_email_requires_confirmation
     fill_in('user[email]', with: 'updated-email@auction.test')
     fill_in('user[current_password]', with: 'password123')
     click_link_or_button('Update')
-    assert(page.has_css?('div.notice', text: 'Updated successfully.'))
+    assert(page.has_css?('div.notice', text: 'Confirmation link was sent to new email address. Please confirm the address for the change to take an effect!'))
```
Comment (artur-beljajev) [inline test/system/users/edit_user_test.rb]: Do we need to rely on HTML structure here? What if we just test it from pure user's perspective?

Like `assert_text 'Confirmation link was sent to new email address. Please confirm the address for the change to take an effect!'`?

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (yulgolem) [reply app/controllers/users_controller.rb]: Yep, and here is the pitfall - we are not assigning attributes here and thus email attribute is not changed.
And we are updating via #update, not saving, and I totally do not want to make a callback in model on update.

File: test/system/users/edit_user_test.rb (hunk 2 above)
Comment (yulgolem) [reply test/system/users/edit_user_test.rb]: Great idea, I'll do that.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (artur-beljajev) [reply app/controllers/users_controller.rb]: What do you think about https://github.com/internetee/auction_center/pull/405/commits/0bc3824309ca92dd4338196d1c80c7fc854fba2b?

Comment (artur-beljajev) [reply app/controllers/users_controller.rb]: I made a mistake, updated.

Comment (yulgolem) [reply app/controllers/users_controller.rb]: I think that's excellent clarification allowing us to use dirty attributes. Thanks a lot!

**Comment by artur-beljajev** [comment]
Thoughts/comments?

https://github.com/internetee/auction_center/pull/405/commits/29678320460e2f9972edf8e6bae2402dd451c86f

**Comment by yulgolem** [comment]
> @yulgolem Thoughts/comments?
> 
> [2967832](https://github.com/internetee/auction_center/commit/29678320460e2f9972edf8e6bae2402dd451c86f) & [a01f5c2](https://github.com/internetee/auction_center/commit/a01f5c278239a087d90dfcf94be598ea9b778d31).

Well, I agree with that. 
I totally dislike to use any callbacks on models, however it can be necessary in some cases, and Active Record surely can silently fail for reasons other than validation errors - AR callbacks that do not return true (eg before_save), invalid parent/ child records etc.

**Pull request merged**